### ISSUE TITLE
Update TensorFlow Image Classifiers samples to use the latest Inception model version

### ISSUE THREAD

**Issue by CESARDELATORRE** [dotnet/machinelearning-samples#574]
Update TensorFlow Image Classifiers samples to use the **latest Inception model version**.

Samples to update:
https://github.com/dotnet/machinelearning-samples/tree/master/samples/csharp/getting-started/DeepLearning_ImageClassification_TensorFlow

https://github.com/dotnet/machinelearning-samples/tree/master/samples/csharp/end-to-end-apps/DeepLearning_ImageClassification_TensorFlow

See:
DIFFERENCES in INCEPTION v3, v2, etc.:
https://datascience.stackexchange.com/questions/15328/what-is-the-difference-between-inception-v2-and-inception-v3

See:

.pb model downloads:

Inception v1 (TensorFlow frozen .pb):
https://storage.googleapis.com/download.tensorflow.org/models/inception5h.zip 

Inception v3 (TensorFlow frozen .pb):
https://storage.googleapis.com/download.tensorflow.org/models/inception_dec_2015.zip

https://storage.googleapis.com/download.tensorflow.org/models/inception_v3_2016_08_28_frozen.pb.tar.gz

https://tfhub.dev/google/imagenet/inception_v3/feature_vector/3

- name: "googlenet-v3" description: "GoogleNet v3 (Inception v3) (https://arxiv.org/pdf/1512.00567.pdf)" model_hash: f5ce67052687aba5bc35250b51e8d0f127d3fd140eac67b47c07d012d20d55a3 tar: https://storage.googleapis.com/download.tensorflow.org/models/inception_v3_2016_08_28_frozen.pb.tar.gz output: "classification/googlenet/v3/tf/" model_path_prefix: "inception_v3_2016_08_28_frozen.pb" model_optimizer_args: "--framework tf --data_type FP32 --reverse_input_channels --input_shape [1,299,299,3] --input input --scale_values input[254.99997577500233] --output InceptionV3/Predictions/Softmax --input_model <googlenet-v3.pb>"
framework: tf 
license: https://github.com/tensorflow/models/blob/master/LICENSE

From <https://www.twblogs.net/a/5c0ab2b3bd9eee6fb37be56e> 

Inception v4 (TensorFlow PYTHON model):
https://github.com/tensorflow/models/blob/master/research/slim/nets/inception_v4.py

Inception v1 (CAFFE):
name: "googlenet-v1" description: "GoogleNet v1 (Inception v1) (https://arxiv.org/pdf/1409.4842.pdf)" model: https://raw.githubusercontent.com/BVLC/caffe/88c96189bcbf3853b93e2b65c7b5e4948f9d5f67/models/bvlc_googlenet/deploy.prototxt model_hash: fa36cd4112a3240a29600a5aa35333f31bec33daa1a6085b0af308a9f50f0c50 weights: http://dl.caffe.berkeleyvision.org/bvlc_googlenet.caffemodel weights_hash: 6f7101e3a2183738a7125a0c5021ba82a1feb4228c5ca0924d991b6daf6f6fad output: "classification/googlenet/v1/caffe" old_dims: [10,3,224,224] new_dims: [1,3,224,224] model_optimizer_args: "--framework caffe --data_type FP32 --input_shape [1,3,224,224] --input data --mean_values data[104.0,117.0,123.0] --output prob --input_model <googlenet-v1.caffemodel> --input_proto <googlenet-v1.prototxt>" framework: caffe license: https://github.com/BVLC/caffe/blob/master/LICENSE

From <https://www.twblogs.net/a/5c0ab2b3bd9eee6fb37be56e> 

Inception v2 (CAFFE):
name: "googlenet-v2" description: "GoogleNet v2 (Inception v2) (https://arxiv.org/pdf/1502.03167.pdf)" model: https://raw.githubusercontent.com/lim0606/caffe-googlenet-bn/d19caf526b7d8cad873ff91ba4cea602eadd58b3/deploy.prototxt model_hash: 18c77f01cca83c4cee973bfd8739968210dd6a1e5af1b857faaec230c5aa7d5a weights: https://github.com/lim0606/caffe-googlenet-bn/raw/d19caf526b7d8cad873ff91ba4cea602eadd58b3/snapshots/googlenet_bn_stepsize_6400_iter_1200000.caffemodel weights_hash: c6581a8e60e5ae8962183ff019986fc9dfacc5e5bd5247d76a0ff7fae8e0409a output: "classification/googlenet/v2/caffe" old_dims: [10,3,224,224] new_dims: [1,3,224,224] layers_to_layer: True model_optimizer_args: "--framework caffe --data_type FP32 --input_shape [1,3,224,224] --input data --mean_values data[104.0,117.0,123.0] --output prob --input_model <googlenet-v2.caffemodel> --input_proto <googlenet-v2.prototxt>" framework: caffe license: https://github.com/lim0606/caffe-googlenet-bn/blob/master/README.md

From <https://www.twblogs.net/a/5c0ab2b3bd9eee6fb37be56e> 

Inception v4 (CAFFE): 
name: "googlenet-v4" description: "GoogleNet v4 (Inception v4)" model: https://drive.google.com/drive/folders/0B9mkjlmP0d7zUEJ3aEJ2b3J0RFU model_hash: 2213d8729839cd015c4b0335d91da1cf4a034627fadedcdb5bf8ac347c4896d9 weights: https://drive.google.com/drive/folders/0B9mkjlmP0d7zUEJ3aEJ2b3J0RFU weights_hash: 6ff248c1215a9fc14ac7ccd44b03da35e41e50bde054ba201bd9c737522996c3 output: "classification/googlenet/v4/caffe" weights_google_drive_id: 0B9mkjlmP0d7zeG1HREVMR2F3WmM model_google_drive_id: 0B9mkjlmP0d7zWEJsRl9zeTQ1NzA weights_size: 170777072 model_size: 86534 model_optimizer_args: "--framework caffe --data_type FP32 --input_shape [1,3,299,299] --input data --mean_values data[128.0,128.0,128.0] --scale_values data[128.0] --output prob --input_model <googlenet-v4.caffemodel> --input_proto <googlenet-v4.prototxt>" framework: caffe license: https://github.com/soeaver/caffe-model/blob/master/LICENSE

From <https://www.twblogs.net/a/5c0ab2b3bd9eee6fb37be56e>

**Comment by DevLob-zz** [comment]
is there is a sample to train and segment image into multiple objects

**Comment by CESARDELATORRE** [comment]
We don't have that sample, currently. We're working on a new ImageClassifier with a high-level API to re-train (transfer learning) which we'll release in a few days as a preview.

About segment image into multiple objects, do you mean object detection like this sample?
https://github.com/dotnet/machinelearning-samples/tree/master/samples/csharp/end-to-end-apps/DeepLearning_ObjectDetection_Onnx

If not, what do you mean? Can you elaborate further on what do you mean with "segment image into multiple objects"?

**Comment by DevLob-zz** [comment]
Yes may be that but one question please do you think this will be applicable with page layout segementaion
